api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 
-	// DiskStorageAccountType is the disk storage account type to use. Valid values are:
-	// * Standard_LRS: HDD
-	// * StandardSSD_LRS: Standard SSD
-	// * Premium_LRS: Premium SDD
-	// * UltraSSD_LRS: Ultra SDD
-	//
-	// Defaults to Premium_LRS. For more details, visit the Azure documentation:
-	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison
+	// diskStorageAccountType is the disk storage account type to use.
+	// Valid values are Standard, StandardSSD, PremiumSSD and UltraSSD and omitted.
+	// Note that Standard means a HDD.
+	// The disk performance is tied to the disk type, please refer to the Azure documentation for further details
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison.
+	// When omitted this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is PremiumSSD.
 	//
-	// +kubebuilder:default:=Premium_LRS
-	// +kubebuilder:validation:Enum=Standard_LRS;StandardSSD_LRS;Premium_LRS;UltraSSD_LRS
+	// +kubebuilder:validation:Enum=Standard;StandardSSD;PremiumSSD;UltraSSD
 	// +optional
+	// TODO: Should all disk options come under a struct rather than be top level like this?
 	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
 
-	// AvailabilityZone is the failure domain identifier where the VM should be attached to. This must not be specified
-	// for clusters in a location that does not support AvailabilityZone.
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Example values are TODO
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
 	//
 	// +optional
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 
-	// DiskEncryptionSetID is the ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs. This
+	// diskEncryptionSetID is the ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs. This
 	// needs to exist in the same subscription id listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.SubscriptionID.
 	// DiskEncryptionSetID should also exist in a resource group under the same subscription id and the same location
 	// listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.Location.
+	// TODO: What does this do for a customer, why would they want to set it?
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
 	//
 	// +optional
 	DiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
 
-	// EnableEphemeralOSDisk is a flag when set to true, will enable ephemeral OS disk.
+	// TODO: This shouldn't be a bool, but I don't know what the field is really for.
```

Comment:
Bools are discourage upstream, and forbidden downstream. I haven't re-implemented this as an enum yet as I have no idea what it does, I was wondering if we create a `Disk` structure, should this be 
```
storageAccountMode: Persistent | Ephemeral
```

Reply:
This was to enable ephemeral OS disks - https://learn.microsoft.com/en-us/azure/virtual-machines/ephemeral-os-disks 

When its enabled we set the caching to read only and the diff disk settings to local - https://github.com/openshift/hypershift/blob/c467b683d6878a23bb8b08ab1a79f3d3e7a160b7/hypershift-operator/controllers/nodepool/azure.go#L90-L89

Reply:
Ok great, so we can explain the use case (lower latency to the disk), the risks, and the side effects, to help the end user understand what's going on with this option. I think we can get all of that from the links you've mentioned